DiscordBot.DataAccess/SheetsServiceRequestsHelper.cs

```diff
@@ -122,6 +127,36 @@ internal static ValueRange MakeCellUpdate(string range, object value)
             };
         }
 
+        internal static T ExecuteRequestsWithRetries<T>(ClientServiceRequest<T> request) =>
+            ExecuteRequestsWithRetriesAsync(request).Result;
+
+        internal static async Task<T> ExecuteRequestsWithRetriesAsync<T>(ClientServiceRequest<T> request)
+        {
+            var random = new Random();
+            var retryMilliseconds = random.Next(1000, 2000);
+
+            while (retryMilliseconds <= 200000)
+            {
+                try
+                {
+                    return await request.ExecuteAsync();
+                }
+                catch (GoogleApiException exception)
+                {
+                    if (exception.HttpStatusCode == HttpStatusCode.TooManyRequests)
+                    {
+                        retryMilliseconds *= 2;
+                        retryMilliseconds += random.Next(1, 1000);
+
+                        Console.WriteLine($"Retry: {retryMilliseconds}");
```

Comment:
Does this write anywhere sensible where you can see it if it's deployed? 🙂

Reply:
Whoops that was for testing! We haven't got any logging set up, so I probably should just remove this? 🙂

Reply:
Yes, that sounds sensible 🙂

Reply:
Removed in 8b6f7c6 🙂